### DISCUSSION TITLE
Install in dist folder to prepare packaging and deployment

### DISCUSSION THREAD

**Discussion by HeneryHawk** [pnpm/pnpm#3591] [Q&A]
Hi,

I have the following setup and a question about it.
My monorepo contains several packages, including a core package, which is used by the others. Each package creates a `dist` folder during the build, in which the package.json of the respective package is copied. Now a `pnpm install --prod --shamefully-hoist` should be executed in this dist folder to install all dependencies so that this folder can be packaged and deployed.

Since I have excluded the dist folders in the `pnpm-workspace.yaml`, the install will not run in the dist folder.
However, I don't want to not exclude the dist folder in `pnpm-workspace.yaml` as it is actually only relevant to the build and also has the same package name as the origin package.

Is there a solution for this?

Regards

**Comment by zkochan** [comment]
Why don't you write the built package to dist and then publish from there using the `publishConfig.directory` setting via `package.json`? 

See https://pnpm.io/package_json#publishconfigdirectory

**Reply by HeneryHawk** [reply]
Hi @zkochan
thanks for your quick reply.

The packages are not published to a registry. A kind of ZIP file is created, which contains all code and all dependencies and is used for the deployment to the environments. Therefore all dependencies should be resolved and contained in the `node_modules`. And the install should be executed only in dist folder and not for the complete repository.

Regards
